File: csharp/lc-896/Solution.cs
namespace lc_896
{
    public class Solution
    {
        public bool IsMonotonic(int[] a)
        {
            bool? trend = null;
            var n = a.Length;
            if (n == 1)
            {
                return true;
            }

            for (var i = 1; i < n; i++)
            {
                var cur = a[i];
                var last = a[i - 1];
                if (cur > last)
                {
                    if (trend == false)
                    {
                        return false;
                    }
                    else
                    {
                        trend = true;
                    }
                }
                else if (cur < last)
                {
                    if (trend == true)
                    {
                        return false;
                    }
                    else
                    {
                        trend = false;
                    }
                }
            }

            return true;
        }
    }
}

File: rust/lc-896/src/solution.rs
pub struct Solution {}

impl Solution {
    pub fn is_monotonic(a: Vec<i32>) -> bool {
        let mut trend = std::cmp::Ordering::Equal;
        let n = a.len();
        if n == 1 {
            return true;
        }

        for (i, cur) in a[1..].iter().enumerate() {
            let diff = a[i].cmp(cur);
            if diff == trend {
                if trend == std::cmp::Ordering::Equal {
                    continue;
                } else {
                    return false;
                }
            } else if trend == std::cmp::Ordering::Equal {
                trend = diff.reverse();
            }
        }

        true
    }
}
